src/__tests__/toggleselectoptions.js

```diff
@@ -0,0 +1,136 @@
+import {render, screen} from '@testing-library/react'
+import React from 'react'
+import userEvent from '..'
+import {setup, addListeners} from './helpers/utils'
+
+test('should fire the correct events for multiple select', () => {
+  const {element: select, getEventCalls} = setup(
+    <select data-testid="element" multiple>
+      <option value="1">1</option>
+      <option value="2">2</option>
+      <option value="3">3</option>
+    </select>,
+  )
+
+  userEvent.toggleSelectOptions(select, '1')
+
+  expect(getEventCalls()).toMatchInlineSnapshot(`
+    mouseover: Left (0)
+    mousemove: Left (0)
+    mousedown: Left (0)
+    focus
+    mouseup: Left (0)
+    click: Left (0)
+    mouseover: Left (0)
+    mousemove: Left (0)
+    mousedown: Left (0)
+    mouseup: Left (0)
+    click: Left (0)
+    change
+  `)
+})
+
+test('should fire the correct events for multiple select when focus is in other element', () => {
+  const {element: select, getEventCalls} = setup(
+    <>
+      <select multiple>
+        <option value="1">1</option>
+        <option value="2">2</option>
+        <option value="3">3</option>
+      </select>
+      <button data-testid="other" tabIndex={0}>
+        Other
+      </button>
+    </>,
+  )
+
+  const getButtonEvents = addListeners(screen.getByTestId('other'))
+
+  screen.getByTestId('other').focus()
+
+  userEvent.toggleSelectOptions(select, '1')
+
+  expect(getEventCalls()).toMatchInlineSnapshot(`
+    mouseover: Left (0)
+    mousemove: Left (0)
+    mousedown: Left (0)
+    focus
+    mouseup: Left (0)
+    click: Left (0)
+    mouseover: Left (0)
+    mousemove: Left (0)
+    mousedown: Left (0)
+    mouseup: Left (0)
+    click: Left (0)
+    change
+  `)
+  expect(getButtonEvents()).toMatchInlineSnapshot(`
+    focus
+    mousemove: Left (0)
+    mouseleave: Left (0)
+    blur
+  `)
```

Comment:
Let's switch the order to make it resemble the order of events in general.

```suggestion
  expect(getButtonEvents()).toMatchInlineSnapshot(`
    focus
    mousemove: Left (0)
    mouseleave: Left (0)
    blur
  `)
  expect(getEventCalls()).toMatchInlineSnapshot(`
    mouseover: Left (0)
    mousemove: Left (0)
    mousedown: Left (0)
    focus
    mouseup: Left (0)
    click: Left (0)
    mouseover: Left (0)
    mousemove: Left (0)
    mousedown: Left (0)
    mouseup: Left (0)
    click: Left (0)
    change
  `)
```